src/main/java/org/springframework/data/couchbase/repository/query/StringBasedCouchbaseQuery.java

```diff
@@ -0,0 +1,103 @@
+import org.springframework.data.repository.query.QueryMethodEvaluationContextProvider;
+import org.springframework.expression.spel.standard.SpelExpressionParser;
+import org.springframework.util.Assert;
+
+/**
+ * Query to use a plain JSON String to create the {@link Query} to actually execute.
+ *
+ * @author Michael Reiche
+ * @since 4.1
+ */
+public class StringBasedCouchbaseQuery extends AbstractCouchbaseQuery {
+
+	private static final String COUNT_EXISTS_AND_DELETE = "Manually defined query for %s cannot be a count and exists or delete query at the same time!";
+	private static final Logger LOG = LoggerFactory.getLogger(StringBasedCouchbaseQuery.class);
+
+	private final SpelExpressionParser expressionParser;
+	private final QueryMethodEvaluationContextProvider evaluationContextProvider;
+	private final NamedQueries namedQueries;
+
+	/**
+	 * Creates a new {@link StringBasedCouchbaseQuery} for the given {@link String}, {@link CouchbaseQueryMethod},
+	 * {@link CouchbaseOperations}, {@link SpelExpressionParser} and {@link QueryMethodEvaluationContextProvider}.
+	 *
+	 * @param method must not be {@literal null}.
+	 * @param couchbaseOperations must not be {@literal null}.
+	 * @param expressionParser must not be {@literal null}.
+	 * @param evaluationContextProvider must not be {@literal null}
+	 * @param namedQueries
+	 */
+	public StringBasedCouchbaseQuery(CouchbaseQueryMethod method, CouchbaseOperations couchbaseOperations,
+			SpelExpressionParser expressionParser, QueryMethodEvaluationContextProvider evaluationContextProvider,
+			NamedQueries namedQueries) {
+
+		super(method, couchbaseOperations, expressionParser, evaluationContextProvider);
+
+		Assert.notNull(expressionParser, "SpelExpressionParser must not be null!");
+
+		this.expressionParser = expressionParser;
+		this.evaluationContextProvider = evaluationContextProvider;
+
+		if (hasAmbiguousProjectionFlags(isCountQuery(), isExistsQuery(), isDeleteQuery())) {
+			throw new IllegalArgumentException(String.format(COUNT_EXISTS_AND_DELETE, method));
+		}
+		this.namedQueries = namedQueries;
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * @see org.springframework.data.couchbase.repository.query.AbstractCouchbaseQuery#createQuery(org.springframework.data.couchbase.repository.query.ConvertingParameterAccessor)
+	 */
+	@Override
+	protected Query createQuery(ParametersParameterAccessor accessor) {
+
+		StringN1qlQueryCreator creator = new StringN1qlQueryCreator(accessor, getQueryMethod(),
+				getOperations().getConverter(), getOperations().getBucketName(), expressionParser, evaluationContextProvider,
+				namedQueries);
+		Query query = creator.createQuery();
+
+		if (LOG.isDebugEnabled()) {
+			LOG.debug("Created query "+ query.export());
+		}
+
+		return query;
+	}
+
+	@Override
+	protected Query createCountQuery(ParametersParameterAccessor accessor) {
+		return /*applyQueryMetaAttributesWhenPresent*/(createQuery(accessor)); // not yet implemented
```

Comment:
should this throw an exception for now, or can we delegate to somewhere? if we leave it like this, would the refactor reduce the features we had (breaking code?)

Reply:
I've added the implementation from Couchbase. It handles a "Meta" annotation (which will be ignored by Couchbase for now).